tunnel/tunnel.go

```diff
@@ -15,94 +10,25 @@ import (
 	"time"
 
 	"github.com/cypherpunkarmory/punch/backoff"
+	"github.com/fatih/color"
 	log "github.com/sirupsen/logrus"
 	"github.com/tj/go-spin"
 	"golang.org/x/crypto/ssh"
-	"golang.org/x/crypto/ssh/terminal"
 )
 
-// From https://sosedoff.com/2015/05/25/ssh-port-forwarding-with-go.html
-// Handle local client connections and tunnel data to the remote server
-// Will use io.Copy - http://golang.org/pkg/io/#Copy
-func handleClient(client io.ReadWriteCloser, remote io.ReadWriteCloser) {
-	defer client.Close()
-	defer remote.Close()
-
-	chDone := make(chan bool)
-	// Start remote -> local data transfer
-	go func() {
-		amt, err := io.Copy(client, remote)
-		if err != nil {
-			log.Debugf("Copy Error: %s ", err)
-		}
-		log.Debugf("Local -> Remote (%d bytes)", amt)
-		chDone <- true
-	}()
-
-	// Start local -> remote data transfer
-	go func() {
-		amt, err := io.Copy(remote, client)
-		if err != nil {
-			log.Debugf("Copy Error %s ", err)
-		}
-		log.Debugf("Local <- Remote (%d bytes)", amt)
-		chDone <- true
-	}()
-	<-chDone
-}
-
-func privateKeyFile(path string) (ssh.AuthMethod, error) {
-	log.Debugf("Parsing privatekey %s", path)
-
-	buffer, err := ioutil.ReadFile(path)
-	if err != nil {
-		return nil, errors.New("cannot read SSH key file " + path)
-	}
-	if len(buffer) == 0 {
-		return nil, errors.New("bad key file empty file")
-	}
-
-	block, _ := pem.Decode(buffer)
-	if block == nil {
-		return nil, errors.New("bad key file")
-	}
-
-	// Return early if the SSH file is not password protected
-	if !x509.IsEncryptedPEMBlock(block) {
-		key, errParse := ssh.ParsePrivateKey(buffer)
-		if errParse != nil {
-			return nil, errors.New("cannot parse SSH key file " + path)
-		}
-		return ssh.PublicKeys(key), nil
-	}
-
-	fmt.Println("Your password: ")
-
-	bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
-	if err != nil {
-		return nil, errors.New("could not read your password " + err.Error())
-	}
-
-	key, err := ssh.ParsePrivateKeyWithPassphrase(buffer, bytePassword)
-	if err != nil {
-		return nil, errors.New("cannot parse SSH key file " + path)
-	}
-
-	return ssh.PublicKeys(key), nil
-}
+var done = "done"
+var starting = "starting"
```

Comment:
You should also define `error` here, at the very least. In my OOP-biased opinion, I think a `TunnelStatus` struct defined somewhere such that was publicly accessible with the values as fields to it would make things even more readable. Then the code would become e.g. `tunnelStatus = TunnelStatus.done` or whatever.